snakemake/shell.py

```diff
@@ -134,6 +142,15 @@ def __new__(
         if conda_env:
             cmd = Conda(container_img).shellcmd(conda_env, cmd)
 
+        tmpdir = None
+        if len(cmd.replace("'", r"'\''")) + 2 > MAX_ARG_LEN:
```

Comment:
Awkward,  but no good alternative without changing `singularity.shellcmd` synopsis. Should get a useful actual length for the script as an argument.